### PR TITLE
accept a string for key_mgmt parameter

### PR DESCRIPTION
**Pull request by kyleaa**
I'm making a (potentially terrible) attempt at this after the comments in Nerves Training. It seemed better to handle this here than in `nerves_network`? Interestingly the docs all referenced using `:WPA_PSK` (which does not work) instead of `:"WPA-PSK"`, so I didn't know that it was worth mentioning the ability to pass an atom as well as a string.

### CODE REVIEW

**Comment by fhunleth** [comment]
Yay!!! Thank you so much for submitting this! 

I like the idea of making the change in this project. I'll make another pass over the PR soon when I have a fresher brain.

File: lib/nerves_wpa_supplicant/messages.ex [hunk 1]
```diff
@@ -17,6 +17,9 @@ defmodule Nerves.WpaSupplicant.Messages do
   def encode(cmd) when is_atom(cmd) do
     to_string(cmd)
   end
+  def encode({:SET_NETWORK, net_id, :key_mgmt, string}) when is_binary(string) do
+    "SET_NETWORK #{net_id} key_mgmt #{string}"
```
Comment (GregMefford) [inline lib/nerves_wpa_supplicant/messages.ex]: I think we could maintain consistency with the rest of the code if we just convert the one thing and depend on the rest of the code to do what it does already:

```elixir
def encode({:SET_NETWORK, net_id, :key_mgmt, key_mgmt}) when is_string(key_mgmt) do
  encode({:SET_NETWORK, net_id, :key_mgmt, String.to_atom(key_mgmt)})
end
```

File: test/wpa_encode_test.exs [hunk 2]
```diff
@@ -42,7 +42,7 @@ defmodule WpaEncodeTest do
     assert Messages.encode(:ADD_NETWORK) == "ADD_NETWORK"
     assert Messages.encode({:SET_NETWORK, 1, :ssid, "SSID"}) == "SET_NETWORK 1 ssid \"SSID\""
     assert Messages.encode({:SET_NETWORK, 1, :psk, "SSID"}) == "SET_NETWORK 1 psk \"SSID\""
-    assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, "SSID"}) == "SET_NETWORK 1 key_mgmt \"SSID\""
+    assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, "SSID"}) == "SET_NETWORK 1 key_mgmt SSID"
```
Comment (GregMefford) [inline test/wpa_encode_test.exs]: I think we should change these assertions to validate the actual things we expect, like this:

```elixir
assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, :"WPA-PSK"}) == "SET_NETWORK 1 key_mgmt WPA-PSK"
assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, "WPA-PSK"}) == "SET_NETWORK 1 key_mgmt WPA-PSK"
assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, :NONE}) == "SET_NETWORK 1 key_mgmt NONE"
assert Messages.encode({:SET_NETWORK, 1, :key_mgmt, "NONE"}) == "SET_NETWORK 1 key_mgmt NONE"
```

File: lib/nerves_wpa_supplicant.ex [hunk 3]
```diff
@@ -82,7 +82,7 @@ defmodule Nerves.WpaSupplicant do
       Option                | Description
       ----------------------|------------
       :ssid                 | Network name. This is mandatory.
-      :key_mgmt             | The security in use. This is mandatory. Set to :NONE, :WPA_PSK
+      :key_mgmt             | The security in use. This is mandatory. Set to NONE or WPA-PSK
```
Comment (GregMefford) [inline lib/nerves_wpa_supplicant.ex]: I think we should be explicit that the "right" thing to do is either `:NONE` or `"WPA-PSK"`, even though `"NONE"` and `:"WPA-PSK"` would also work.

**Review by GregMefford — COMMENTED** [review]
Thanks so much for taking the time to make this PR! Lots of people have complained about the usability of this API and we just hadn't gotten around to coming up with a solution yet. 🎈 🚀

**Comment by kyleaa** [comment]
Makes total sense, I addressed the comments in the latest commit.

**Review by GregMefford — APPROVED** [review]
I haven't tried actually running it yet, but it looks good to me! 👍

**Comment by kyleaa** [comment]
Just started up a new nerves project and ran into this again - if this is good to go would someone be willing to merge? 😀

**Comment by ConnorRigby** [comment]
@kyleaa Are you using this lib directly? I think we handle this in `nerves_network` now, but this should probably be the correct place for it

**Review by ConnorRigby — APPROVED** [review]

**Comment by kyleaa** [comment]
Whoops, this probably wasn’t the cause of my woes (what i get for changing multiple things at once), but it reminded me to check and see that this was never merged.

**Review by fhunleth — APPROVED** [review]

**Comment by ConnorRigby** [comment]
This is now fixed at the `nerves_network` level now. I'm open to this change, but i think it shouldn't be limited to just the key_mgmt field. If we go down this route, i think we should handle _every_ field.

**Comment by ConnorRigby** [comment]
As an update to this:
`:wep_key0..3` need to all be atoms for some reason. Very much related to this. It feel very weird to turn your password into an atom.

**Comment by GregMefford** [comment]
Yeah, let's not turn passwords or SSIDs into atoms. They're not globally static things like `:"WPK-PSK"`, so I think it doesn't make sense.

**Comment by ConnorRigby** [comment]
I took a look at the C code that talks to `wpa_ctrl` here. I really want to change it so all keys (ie the left side of the `=` in wpa_supplicant.conf are atoms, and the right side are strings or numbers. Too much of that is handled at the `nerves_network` layer and i do believe they should be handled here.

**Comment by fhunleth** [comment]
Closing old issue. If you haven't already moved to `vintage_net`, please do. It is friendlier on issues like this.

**Pull request closed without merging**